### PR TITLE
Improving test coverage for Response methods

### PR DESCRIPTION
**Pull request by Rohith-Raju**
#186 
Signed-off-by: Rohith-Raju <rohithraju488@gmail.com>

### CODE REVIEW

**Comment by Rohith-Raju** [comment]
@gavv I'm using the latest version of go and there seems to be a problem. Let me downgrade

**Comment by gavv** [comment]
Hi, you can use any version of ho locally, but the code should be compatible with go 1.14 because we still support it.

**Comment by Rohith-Raju** [comment]
@gavv is this approach acceptable?

**Comment by gavv** [comment]
The test itself looks fine. We need to change two things though:

* test NewResponse instead of newResponse
* extract it into separate test, because TestResponseFailed is for testing behavior of Response when it's created with a failed chain

Let me know when it will be ready for review.

**Comment by gavv** [comment]
Please rebase on fresh `master`.

**Comment by Rohith-Raju** [comment]
@gavv  I couldn't use the NewResponse because it doesn't accept an array of rtt  args

**Comment by gavv** [comment]
> @gavv I couldn't use the NewResponse because it doesn't accept an array of rtt .

Actually it does:

```
func NewResponse(
	reporter Reporter, response *http.Response, rtt ...time.Duration,
) *Response {
```

**Comment by Rohith-Raju** [comment]
> 

Ok sure, let me retry

**Comment by Rohith-Raju** [comment]
> > @gavv I couldn't use the NewResponse because it doesn't accept an array of rtt .
> 
> Actually it does:
> 
> ```
> func NewResponse(
> 	reporter Reporter, response *http.Response, rtt ...time.Duration,
> ) *Response {
> ```
Ok sure let me retry

**Comment by Rohith-Raju** [comment]
@gavv is my approach fine ? should I add all the `AssertUsage` tests in one pr?

File: response_test.go [hunk 1]
```diff
@@ -85,6 +85,18 @@ func TestResponseFailed(t *testing.T) {
 	})
 }
 
+func TestMultipleRttArgs(t *testing.T) {
+	check := func(resp *Response) {
+		resp.chain.assertFailed(t)
+	}
```
Comment (gavv) [inline response_test.go:91]: I think this check() function is unnecessary, we can inline it.

**Comment by gavv** [comment]
Thanks for update.

> @gavv is my approach fine ?

Yep.

> should I add all the AssertUsage tests in one pr?

I think yes, why not.

**Comment by gavv** [comment]
Also I think it makes sense to merge all usage tests into one test with sub-tests.

File: response_test.go (hunk 1 above)
Comment (Rohith-Raju) [reply response_test.go:91]: Ok sure @gavv

**Comment by Rohith-Raju** [comment]
> Also I think it makes sense to merge all usage tests into one test with sub-tests.

On it!!

**Comment by Rohith-Raju** [comment]
@gavv done with the changes.

**Comment by gavv** [comment]
Thanks!

Small follow-up commit: 82cc2f171b292c49fb73c4dcb109b8647ae8ea75

**Pull request merged**